### ISSUE TITLE
Error : /*!999999- enable the sandbox mode */

### ISSUE THREAD

**Issue by manunord** [Dolibarr/dolibarr#29724]
### Bug

MariaDB 10.5.25, 10.6.18, 10.11.8, 11.0.6, 11.1.5, 11.2.4 and 11.4.2  now has an option --sandbox
https://mariadb.org/mariadb-dump-file-compatibility-change/
So we have an error when launching a backup with mysqldump: Error : /*!999999- enable the sandbox mode */

The backup created is valid but with an .err extension

### Dolibarr Version

All

### Environment PHP

php 7 and php 8

### Environment Database

MariaDB 10.5.25

### Attached files

![Capture104130](https://github.com/Dolibarr/dolibarr/assets/66733428/8c249617-c707-4dc7-a1a6-2120788d946a)

**Comment by georiane** [comment]
Bonjour,

Quel est votre hébergeur de Dolibarr ?

J'ai ouvert une dissuasion sur www.dolibarr.fr/forum/t/erreur-sauvegarde-base-de-donnee/46460/5   sans solution pour l'instant


Hello,

Who is your Dolibarr host?

I opened a dissuasion on www.dolibarr.fr/forum/t/ERROR-sauvegarde-base-de-donnee/46460/5 without solution for the moment

**Comment by amityweb** [comment]
I dont think this is specific to Dolibarr. I have no idea what Dolibarr is, but when I run a standard mysqldump on my Drupal database, it adds this to the top, which then causes an error import. So any automated database migrations we have fail and need manual intervention. But its not Dolibarr. 

More issues here with links to Maria DB changes https://github.com/ddev/ddev/issues/6249.

**Comment by daraelmin** [comment]
It is related to mariadb and mysql incompatibility 

https://mariadb.org/mariadb-dump-file-compatibility-change/

**Comment by jman482002** [comment]
Same problem here. v19.02
How to fix?

**Comment by georiane** [comment]
Bonjour,

Sur le site [www.dolibarr.fr/forum/t/erreur-sauvegarde-base-de-donnee/46460/34](http://www.dolibarr.fr/forum/t/erreur-sauvegarde-base-de-donnee/46460/34)  les réponses 34/37, 35/37 et 36/37 donne des solutions.

34/37 Modifier utils.class.php

$fullcommandcrypted = $command." ".$paramcrypted." 2>&1";
$fullcommandclear = $command." ".$paramclear." 2>&1";

Par 

$fullcommandcrypted = $command." ".$paramcrypted."|tail +2 2>&1";
$fullcommandclear = $command." ".$paramclear."|tail +2 2>&1";

35/37 Modifier utils.class.php

modifié (la ligne environ 530)
if ($ok && preg_match('/^-- (MySql|MariaDB)/i', $errormsg)) {	// No error
En ajoutant la condition supplémentaire
(str_contains($errormsg, '!999999\- enable the sandbox mode'))

Ce qui donne
if (($ok && preg_match('/^-- (MySql|MariaDB)/i', $errormsg)) || (str_contains($errormsg, '!999999\- enable the sandbox mode'))) {	// No error

36/37 Dolibarr v20 saura gérer cette évolution mariadb.

Hello,

On the site [www.dolibarr.fr/forum/t/erreur-sauvegarde-base-de-donnee/46460/34](http://www.dolibarr.fr/forum/t/erreur-sauvegarde-base-de-donnee/46460/34)  the answers 34/37, 35/37 and 36/37 give solutions.

34/37 Edit utils.class.php

$fullcommandcrypted = $command." ".$paramcrypted." 2>&1";
$fullcommandclear = $command." ".$paramclear." 2>&1";

By

$fullcommandcrypted = $command." ".$paramcrypted."|tail +2 2>&1";
$fullcommandclear = $command." ".$paramclear."|tail +2 2>&1";

35/37 Edit utils.class.php

modified (line approximately 530)
if ($ok && preg_match('/^-- (MySql|MariaDB)/i', $errormsg)) {	// No error
By adding the additional condition 
(str_contains($errormsg, '!999999\- enable the sandbox mode'))

Which give
if (($ok && preg_match('/^-- (MySql|MariaDB)/i', $errormsg)) || (str_contains($errormsg, '!999999\- enable the sandbox mode'))) {	// No error

36/37 Dolibarr v20 will be able to manage this mariadb evolution.

**Comment by hregis** [comment]
another info for the newbe...

In MariaDB 10.4 and later, the mysql.global_priv table has replaced the mysql.user table, and mysql.user should be considered obsolete.
It is now a view into mysql.global_priv created for compatibility with older applications and monitoring scripts.

for the french people : 

Dans MariaDB 10.4 et versions ultérieures, la table mysql.global_priv a remplacé la table mysql.user et mysql.user doit être considérée comme obsolète.
Il s'agit désormais d'une vue dans mysql.global_priv créée pour la compatibilité avec les anciennes applications et scripts de surveillance.
